src/geojson.rs

```diff
@@ -0,0 +1,189 @@
+                    f64::from(node.node()?.decimicro_lat) / 10_000_000.0,
+                ])
+            })
+            .collect()
+    };
+
+    let linestrings: Vec<Vec<geojson::Position>> = obj
+        .relation()?
+        .refs
+        .iter()
+        .filter_map(|child: &Ref| {
+            // todo treat 'inner' and contained relations as well
+            if matches!(child.role.as_str(), "outer") {
+                Some(to_coords(all_objs.get(&child.member)?.way()?)?)
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    // todo report missing geometry or broken linering
+    let mut linering = create_continuous_linering(&linestrings).ok()?;
+
+    // respect right hand rule
+    if is_clockwise(&linering) {
+        linering.reverse();
+    }
+
+    Some(geojson::Value::Polygon(vec![linering]))
+}
+
+/// create a continuous ring from line strings
+fn create_continuous_linering(
+    linestrings: &Vec<Vec<geojson::Position>>,
+) -> Result<Vec<geojson::Position>> {
+    if linestrings.is_empty() || linestrings.iter().any(|ls| ls.len() < 2) {
+        bail!("No linestrings or a linestring has less than 2 positions")
+    }
+
+    // Convert the endpoint positions to a hashable type (tuple) and build the index map
+    let mut endpoints: HashMap<FloatTuple, Vec<usize>> = HashMap::new();
+    for (i, linestring) in linestrings.iter().enumerate() {
+        let start = FloatTuple(
+            linestring.first().context("Empty linestring")?[0],
+            linestring.first().unwrap()[1],
+        );
+        let end = FloatTuple(linestring.last().unwrap()[0], linestring.last().unwrap()[1]);
+        endpoints.entry(start).or_default().push(i);
+        if start != end {
+            // Avoid double entry for loops
+            endpoints.entry(end).or_default().push(i);
+        }
+    }
+
+    // Start from the first linestring
+    let first_index = 0;
+    let mut continuous_line = linestrings[first_index].clone();
+
+    // Track used linestrings to prevent infinite loops
+    let mut used_linestrings = HashSet::new();
+    used_linestrings.insert(first_index);
+
+    while used_linestrings.len() < linestrings.len() {
+        let current_end_key = FloatTuple(
+            continuous_line.last().context("Empty line ring")?[0],
+            continuous_line.last().unwrap()[1],
+        );
+
+        if let Some(indices) = endpoints.get(&current_end_key) {
+            let Some(&next_index) = indices.iter().find(|&&i| !used_linestrings.contains(&i))
+            else {
+                bail!("No matching linestring found");
+            };
+
+            let next_linestring = &linestrings[next_index];
+            let next_start_key = FloatTuple(next_linestring[0][0], next_linestring[0][1]);
+            let next_end_key = FloatTuple(
+                next_linestring.last().context("Next linestring empty.")?[0],
+                next_linestring.last().unwrap()[1],
+            );
```

Comment:
Inkonsistentes error handling.
```suggestion
            let next_end_key = {
                let x = next_linestring.last().context("Next linestring empty")?;
                FloatTuple(x[0], x[1])
            };
```